doc/ip-interface/loopback-action/ip-interface-loopback-action-design.md

```diff
@@ -0,0 +1,318 @@
+- [1 Introduction](#1-introduction)
+    - [1.1 Feature overview](#11-feature-overview)
+    - [1.2 Requirements](#12-requirements)
+        - [1.2.1 Functionality](#121-functionality)
+        - [1.2.2 Command interface](#122-command-interface)
+        - [1.2.3 Error handling](#123-error-handling)
+        - [1.2.4 Event logging](#124-event-logging)
+- [2 Design](#2-design)
+    - [2.1 Overview](#21-overview)
+    - [2.2 DB schema](#22-db-schema)
+        - [2.2.1 Config DB](#221-config-db)
+        - [2.2.2 App DB](#222-app-db)
+        - [2.2.3 Configuration migration](#223-configuration-migration)
+    - [2.3 Flows](#23-flows)
+        - [2.3.1 Loopback action set interface vlan](#231-loopback-action-set-interface-vlan)
+            - [2.3.1.1 Handle set in runtime](#2311-handle-set-in-runtime)
+            - [2.3.1.2 Handle set in init](#2312-handle-set-in-init)
+        - [2.3.2 Loopback action set interface Ethernet](#232-loopback-action-set-interface-ethernet)
+        - [2.3.3 Loopback action set interface port-channel](#233-loopback-action-set-interface-port-channel)
+        - [2.3.4 IP interface creation](#234-ip-interface-creation)
+    - [2.4 SAI](#24-sai)
+    - [2.5 Statistics](#25-statistics)
+    - [2.6 CLI](#26-cli)
+        - [2.6.1 Config command](#261-config-command)
+        - [2.6.2 Show command](#262-show-command)
+    - [2.7 YANG model](#27-yang-model)
+    - [2.8 Warm/Fast boot](#28-warmfast-boot)
+- [3 Test plan](#3-test-plan)
+    - [3.1 Unit tests via VS](#31-unit-tests-via-vs)
+    - [3.2 System tests](#32-system-tests)
+- [4 Open items](#4-open-items)    
+
+## Revision
+| Rev | Date       | Author         | Description     |
+|:---:|:----------:|:--------------:|:----------------|
+| 0.1 | 18/05/2022 | Lior Avramov   | Initial version |
+
+## Scope
+This document describes the high level design of IP interface loopback action feature in SONiC
+
+## List of figures
+[Figure 1: feature overview](#figure-1-feature-overview)  
+[Figure 2: interface vlan](#figure-2-interface-vlan)  
+
+# 1 Introduction
+## 1.1 Feature overview
+IP interface loopback action is a feature that allows user to change the way router handles routed packets for which egress port equals to ingress port.
+1. When loopback action is configured to drop, those packets will be dropped.
+2. When loopback action is configured to forward, those packets will be forwarded.
+
+![feature overview](images/feature_overview.svg "Figure 1: feature overview")
+###### Figure 1: feature overview
+
+## 1.2 Requirements
+### 1.2.1 Functionality
+**This feature will support the following functionality:**
+1. Loopback action can be set on IP interface. 
+2. IP Interface can be one of the following:
+    1. Interface vlan
+    2. Interface Ethernet
+    3. Interface port-channel 
+3. Loopback action can be set to "drop" or "forward". 
+4. CLI does not allow to set loopback action on non IP interface. Interface is considered as IP interface if it has entry in:
```

Comment:
suggest to rephrase 
 CLI will reject loopback action setting on a non IP interface. Interface is considered as an IP interface if it has entry in